packages/docusaurus-theme-classic/src/theme/NavbarItem/DocSidebarNavbarItem.tsx

```diff
@@ -15,11 +15,19 @@ import {useDocsPreferredVersion, uniq} from '@docusaurus/theme-common';
 import type {GlobalDataVersion} from '@docusaurus/plugin-content-docs';
 
 function getSidebarInVersion(versions: GlobalDataVersion[], sidebarId: string) {
-  const allSidebars = versions.flatMap((version) => version.sidebars);
+  const allSidebars = versions.map((version) => version.sidebars);
```

Comment:
I'd rather use a flat 1-level array here. You can create an array of sidebar entries `[sidebarId,sidebar]`

```js
  const allSidebars = versions.flatMap((version) => Object.entries(version.sidebars));
```

This makes the following code simpler:

```js
const sidebarEntry = allSidebars.find((sidebarEntry) => sidebarEntry[0] === sidebarId)
```